bofire/strategies/shortest_path.py

```diff
@@ -0,0 +1,100 @@
+from typing import Optional, Tuple
+
+import cvxpy as cp
+import numpy as np
+import pandas as pd
+
+from bofire.data_models.constraints.api import (
+    LinearConstraint,
+    LinearEqualityConstraint,
+    LinearInequalityConstraint,
+)
+from bofire.data_models.domain.api import Constraints, Inputs
+from bofire.data_models.features.api import ContinuousInput
+from bofire.data_models.strategies.api import ShortestPathStrategy as DataModel
+from bofire.strategies.strategy import Strategy
+
+
+class ShortestPathStrategy(Strategy):
+    def __init__(
+        self,
+        data_model: DataModel,
+        **kwargs,
+    ):
+        self.start = pd.Series(data_model.start)
+        self.end = pd.Series(data_model.end)
+        self.atol = data_model.atol
+        super().__init__(data_model=data_model, **kwargs)
+
+    @property
+    def continuous_inputs(self) -> Inputs:
+        return self.domain.inputs.get(ContinuousInput)  # type: ignore
+
+    def get_linear_constraints(
+        self, constraints: Constraints
+    ) -> Tuple[np.ndarray, np.ndarray]:
+        inputs = self.continuous_inputs
+        keys = inputs.get_keys()
+        b = np.array([c.rhs for c in constraints])  # type: ignore
+        A = np.zeros([len(constraints), len(inputs)])
+        for i, c in enumerate(constraints):
+            assert isinstance(c, LinearConstraint)
+            for key, coef in zip(c.features, c.coefficients):
+                feat = inputs.get_by_key(key)
+                assert isinstance(feat, ContinuousInput)
+                if feat.is_fixed():
+                    b[i] -= feat.fixed_value()[0] * coef  # type: ignore
+                else:
+                    A[i, keys.index(key)] = coef
+        return A, b
+
+    def step(self, start: pd.Series) -> pd.Series:
+        inputs = self.continuous_inputs
```

Comment:
Can you write a doc-string for this method?

Reply:
Will do it.

Reply:
Done, since Monday, I have copilot access for a trial period. It is super powerful for documentation.